### DISCUSSION TITLE
yutto如何下载Bilibili付费课程

### DISCUSSION THREAD

**Discussion by Nightingale0504** [yutto-dev/yutto#233] [Q&A] [answered]
如题

**Comment by Nightingale0504** [comment]
![Snipaste_2024-02-15_02-17-08](https://github.com/yutto-dev/yutto/assets/61570274/ea9df12d-163f-48fc-bfee-ebdf56778f5a)
我已购买的付费课程下载失败

**Reply by lc4t** [reply]
`yutto 'https://www.bilibili.com/cheese/play/ss5918' -c '....' -b`  这样是可以的，你下载时用的命令是什么？

**Comment by Nightingale0504** [comment]
> `yutto 'https://www.bilibili.com/cheese/play/ss5918' -c '....' -b` 这样是可以的，你下载时用的命令是什么？
yutto -b --no-danmaku -c "****** https://www.bilibili.com/cheese/play/ss1525
PS: ******为我真实cookie

**Reply by lc4t** [reply]
贴一下你浏览器直接访问`https://api.bilibili.com/x/player/v2?cid=1051169484&aid=865363180`的返回？你是不是非大陆IP..？

**Reply by Nightingale0504** [reply]
{
    "code": 0,
    "message": "0",
    "ttl": 1,
    "data": {
        "aid": 865363180,
        "bvid": "BV1F54y1u737",
        "allow_bp": false,
        "no_share": false,
        "cid": 1051169484,
        "max_limit": 3000,
        "page_no": 1,
        "has_next": false,
        "ip_info": {
            "ip": "183.191.131.164",
            "zone_ip": " 10.66.22.28",
            "zone_id": 4456464,
            "country": "中国",
            "province": "山西",
            "city": "太原"
        },
        "login_mid": 545844570,
        "login_mid_hash": "2ecc05d2",
        "is_owner": false,
        "name": "水月如星空般深蓝",
        "permission": "10000,1001",
        "level_info": {
            "current_level": 5,
            "current_min": 10800,
            "current_exp": 17114,
            "next_exp": 28800,
            "level_up": 1671189147
        },
        "vip": {
            "type": 2,
            "status": 1,
            "due_date": 1739116800000,
            "vip_pay_type": 1,
            "theme_type": 0,
            "label": {
                "path": "",
                "text": "年度大会员",
                "label_theme": "annual_vip",
                "text_color": "#FFFFFF",
                "bg_style": 1,
                "bg_color": "#FB7299",
                "border_color": "",
                "use_img_label": true,
                "img_label_uri_hans": "",
                "img_label_uri_hant": "",
                "img_label_uri_hans_static": "https://i0.hdslb.com/bfs/vip/8d4f8bfc713826a5412a0a27eaaac4d6b9ede1d9.png",
                "img_label_uri_hant_static": "https://i0.hdslb.com/bfs/activity-plat/static/20220614/e369244d0b14644f5e1a06431e22a4d5/VEW8fCC0hg.png"
            },
            "avatar_subscript": 1,
            "nickname_color": "#FB7299",
            "role": 3,
            "avatar_subscript_url": "",
            "tv_vip_status": 0,
            "tv_vip_pay_type": 0,
            "tv_due_date": 0,
            "avatar_icon": {
                "icon_type": 1,
                "icon_resource": {}
            }
        },
        "answer_status": 0,
        "block_time": 0,
        "role": "0",
        "last_play_time": 0,
        "last_play_cid": 0,
        "now_time": 1707935798,
        "online_count": 1,
        "need_login_subtitle": false,
        "subtitle": {
            "allow_submit": false,
            "lan": "",
            "lan_doc": "",
            "subtitles": [
                {
                    "id": 1178939123415670300,
                    "lan": "ai-zh",
                    "lan_doc": "中文（自动生成）",
                    "is_lock": false,
                    "subtitle_url": "//aisubtitle.hdslb.com/bfs/ai_subtitle/prod/8653631801051169484bbb59698ed91add454d6683e94b94f36?auth_key=1707935798-55ca6701243040d29ac264e2b676c082-0-9c374d7b72b98edd7f8fdc60c30f10bc",
                    "type": 1,
                    "id_str": "1178939123415670272",
                    "ai_type": 0,
                    "ai_status": 2
                }
            ]
        },
        "player_icon": {
            "ctime": 1657876281
        },
        "view_points": [],
        "preview_toast": "为创作付费，购买观看完整视频|购买观看",
        "pugv": {
            "watch_status": 3,
            "pay_status": 1,
            "season_status": 1
        },
        "options": {
            "is_360": false,
            "without_vip": false
        },
        "guide_attention": [],
        "jump_card": [],
        "operation_card": [],
        "online_switch": {
            "enable_gray_dash_playback": "500",
            "new_broadcast": "1",
            "realtime_dm": "1",
            "subtitle_submit_switch": "1"
        },
        "fawkes": {
            "config_version": 27690,
            "ff_version": 21289
        },
        "show_switch": {
            "long_progress": false
        },
        "bgm_info": null,
        "toast_block": false,
        "is_upower_exclusive": false,
        "is_upower_play": false,
        "is_ugc_pay_preview": false,
        "elec_high_level": {
            "privilege_type": 0,
            "level_str": "",
            "title": "",
            "intro": ""
        },
        "disable_show_up_info": false
    }
}

**Reply by lc4t** [reply]
这个正常返回，根据报错提示，是你访问这个API的时候`data`键出错了，不过不知道具体你是哪个URL触发的，可能需要你根据执行后的返回判断下是在哪个avid和cid下出错的

**Reply by Nightingale0504** [reply]
我开了debug，这是输出
![Snipaste_2024-02-15_02-41-45](https://github.com/yutto-dev/yutto/assets/61570274/674558a0-5919-4329-baa0-da9292f4f554)

**Reply by Nightingale0504** [reply]
这是正常的DEBUG输出
![Snipaste_2024-02-15_02-43-54](https://github.com/yutto-dev/yutto/assets/61570274/73c0fbbc-ce76-4802-bad0-b0d90d996b47)

**Reply by lc4t** [reply]
`https://api.bilibili.com/x/player/v2?cid=1070693799&aid=354106665` 这个返回好像确实有问题，直接访问是-404，你也可以直接访问看看。
这个视频是哪一个？

**Reply by lc4t** [reply]
哦知道是哪一个了，不过我没购买（

**Reply by lc4t** [reply]
<img width="1416" alt="image" src="https://github.com/yutto-dev/yutto/assets/11334449/920d1310-0006-47f2-867b-a3b55b0b26f5">
看起来是阿B返回非预期了..

**Reply by Nightingale0504** [reply]
是，这个课程我已经买了，但是却下载不了，要向B站反馈吗

**Reply by lc4t** [reply, marked as answer]
反馈阿b也没人理...
加上参数`--no-subtitle`试试？ 如果成功的话你看下视频内容是否正常，这个`--no-subtitle`是不生成字幕文件的意思。

**Reply by Nightingale0504** [reply]
好了，谢谢佬！

**Reply by SigureMo** [reply]
这看起来是 yutto 本身未预期导致的，我稍后加一个检查和 warning，这里按理说不应该直接挂掉
